storage/tests/unit/test_bucket.py

```diff
@@ -535,66 +544,109 @@ def test_create_w_user_project(self):
             bucket.create()
 
     def test_create_w_missing_client_project(self):
+        from google.cloud.storage.client import Client
+
         BUCKET_NAME = "bucket-name"
-        connection = _Connection()
-        client = _Client(connection, project=None)
-        bucket = self._make_one(client, BUCKET_NAME)
+        connection = _make_connection()
+        client = Client(project=None)
 
-        with self.assertRaises(ValueError):
-            bucket.create()
+        with mock.patch(
+            'google.cloud.storage.client.Client._connection',
+            new_callable=mock.PropertyMock
+        ) as client_mock:
+            client_mock.return_value = connection
```

Comment:
Why not just set `client._connection = connection`?

Reply:
I don't remember really, but looks like I've done it because of `_connection.setter`. You right anyway, it can be done much more easier. Rewritten